### PR TITLE
[Java] Implementing inheritance logic in Java

### PR DESCRIPTION
**Pull request by himanshugoyal1065**
#1997

### CODE REVIEW

File: languages/java/exercises/concept/inheritance/.docs/after.md
Comment (mirkoperillo) [inline languages/java/exercises/concept/inheritance/.docs/after.md:18]: missing the fourth type of modifier in Java, the default one, aka `package visibility`

File: languages/java/exercises/concept/inheritance/.docs/after.md
Comment (mirkoperillo) [inline languages/java/exercises/concept/inheritance/.docs/after.md:23]: I see only two concepts described below this sentence

File: languages/java/exercises/concept/inheritance/.docs/introduction.md
Comment (mirkoperillo) [inline languages/java/exercises/concept/inheritance/.docs/introduction.md:1]: I think OOPS is a typo, maybe it was OOP ?

File: languages/java/exercises/concept/inheritance/.docs/introduction.md
Comment (mirkoperillo) [inline languages/java/exercises/concept/inheritance/.docs/introduction.md:47]: Same of above ?
OOPS instead of OOP

File: languages/java/exercises/concept/inheritance/src/test/java/FighterTest.java
Comment (mirkoperillo) [inline languages/java/exercises/concept/inheritance/src/test/java/FighterTest.java:8]: Usually the first test should not be ignored

File: languages/java/exercises/concept/inheritance/src/test/java/FighterTest.java
Comment (mirkoperillo) [inline languages/java/exercises/concept/inheritance/src/test/java/FighterTest.java:15]: Usually we put a message with the `@Ignore` annotation like this
` @Ignore("Remove to run the test")`

**Review by mirkoperillo — CHANGES_REQUESTED** [review]
Thank you @himanshugoyal1065 this is a good work, there are only some little changes to do

After that remember to add an entry into
`languages/java/config.json`

File: languages/java/exercises/concept/inheritance/.docs/after.md
Comment (himanshugoyal1065) [reply languages/java/exercises/concept/inheritance/.docs/after.md:18]: Makes sense. Just didn't want to overwhelm the reader. But now that I think about it, I think we should have it.

File: languages/java/exercises/concept/inheritance/.docs/introduction.md
Comment (himanshugoyal1065) [reply languages/java/exercises/concept/inheritance/.docs/introduction.md:1]: 🤣  we should hide this comment conversation from the people out there. OOP vs OOPS is still an ambiguous topic. 
If you say, I'll make it OOP. Anyways, both of them mean the same!

File: languages/java/exercises/concept/inheritance/.docs/introduction.md
Comment (himanshugoyal1065) [reply languages/java/exercises/concept/inheritance/.docs/introduction.md:47]: Noted!

File: languages/java/exercises/concept/inheritance/src/test/java/FighterTest.java
Comment (himanshugoyal1065) [reply languages/java/exercises/concept/inheritance/src/test/java/FighterTest.java:8]: Oh okay, I'll change that.

File: languages/java/exercises/concept/inheritance/src/test/java/FighterTest.java
Comment (himanshugoyal1065) [reply languages/java/exercises/concept/inheritance/src/test/java/FighterTest.java:15]: yes. Noted.

File: languages/java/exercises/concept/inheritance/.docs/after.md
Comment (himanshugoyal1065) [reply languages/java/exercises/concept/inheritance/.docs/after.md:23]: I think I shouldn't mention aggregation at all.

**Comment by iHiD** [comment]
@himanshugoyal1065 This is awesome work.

A general comment for this repo - it's best not to merge master into this PR, else you end up accidentally requesting reviews off everyone that has code edited on master since you started the PR (You'll notice that you've now accidentally requested a review from @exercism/ruby for example). This is really a bit of a GitHub bug in my opinion, but rebasing rather than merging will avoid the issue 🙂

**Review by mirkoperillo — APPROVED** [review]
@himanshugoyal1065 good work.
There is only a little fix to do before merging it.
If you run `gradle check` in exercise folder you can see that the code breaks some style rules.

**Comment by himanshugoyal1065** [comment]
@mirkoperillo, thank you! I have updated the PR and now it's passing the `gradle check`
Also, I suggest adding this check in the actions directly, so that you or any other maintainer don't have to tell everyone personally :)

**Comment by iHiD** [comment]
> Also, I suggest adding this check in the actions directly
@himanshugoyal1065 Sounds like a good next PR! 😉

**Comment by ynfle** [comment]
@himanshugoyal1065, I implemented this for the `nim` track. [Here](https://github.com/exercism/v3/pull/2237) is the PR if you'd like to take a look if you aren't that familiar with GitHub Actions

**Comment by mirkoperillo** [comment]
@himanshugoyal1065 LGTM, time to merge it. 
Thank you for helping the java track

**Pull request merged**
